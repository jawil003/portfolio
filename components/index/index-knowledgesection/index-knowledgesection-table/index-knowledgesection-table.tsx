import { css } from "@emotion/react";
import React from "react";
import designSystem from "../../../../styles/designSystem";
import { Typography } from "../../../shared/typography";

export interface Props {
  title: string;
  description: string;
  items: string[];
  color?: string;
}

/**
 * An KnowledgeTable React Component.
 * @author Jannik Will
 * @version 0.1
 */
export const IndexKnowledgeTable: React.FC<Props> = ({
  title,
  description,
  items,
  color,
}) => {
  return (
    <div
      css={css`
        & {
          padding: 40px;
          min-height: 430px;
          width: 300px;
          background-color: ${color};
          border-radius: 30px;
          display: flex;
          flex-direction: column;
          align-items: flex-start;
          padding: 40px;
        }
      `}
    >
      <div>
        <Typography
          color={
            designSystem.colors.brand
              .primary
          }
          variant="h5"
        >
          {title}
        </Typography>
        <Typography
          css={css`
            & {
              margin: 0px;
            }
          `}
          variant="b2"
          color={
            designSystem.colors.brand
              .primary
          }
        >
          {description}
        </Typography>
      </div>
      <ul
        css={css`
          & {
            padding-left: 20px;
          }
        `}
      >
        {items.map((i) => (
          <li
            css={css`
              & {
                color: ${designSystem
                  .colors.brand
                  .primary};
              }
            `}
            key={i}
          >
            {i}
          </li>
        ))}
      </ul>
    </div>
  );
};

IndexKnowledgeTable.defaultProps = {
  color:
    designSystem.colors.brand.secondary,
};
