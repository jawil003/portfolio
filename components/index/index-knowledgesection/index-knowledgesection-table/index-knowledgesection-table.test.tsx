import { render } from "@testing-library/react";
import React from "react";
import { IndexKnowledgeTable } from "./index-knowledgesection-table";

describe("IndexKnowledgeTable", () => {
  test("Create IndexKnowledgeTable", () => {
    const testIndexKnowledgeTable = render(
      <IndexKnowledgeTable
        title="test"
        description="Lorem ipsum"
        items={["test", "test1"]}
      />,
    );

    expect(
      testIndexKnowledgeTable.baseElement,
    ).toMatchSnapshot();
  });
});
