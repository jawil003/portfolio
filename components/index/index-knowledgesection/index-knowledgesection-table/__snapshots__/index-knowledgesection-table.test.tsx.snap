// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexKnowledgeTable Create IndexKnowledgeTable 1`] = `
.emotion-0 {
  padding: 40px;
  min-height: 430px;
  width: 300px;
  background-color: #00739d;
  border-radius: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 40px;
}

.emotion-1 {
  text-align: start;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  letter-spacing: normal;
  word-spacing: normal;
  color: #f3f3f3;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-1 {
  font-weight: 400;
  font-size: 1.5em;
}

.emotion-3 {
  text-align: start;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  letter-spacing: normal;
  word-spacing: normal;
  color: #f3f3f3;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-3 {
  font-weight: 400;
  font-size: 0.875em;
}

.emotion-3 {
  margin: 0px;
}

.emotion-5 {
  padding-left: 20px;
}

.emotion-6 {
  color: #f3f3f3;
}

<body>
  <div>
    <div
      class="emotion-0"
    >
      <div>
        <h5
          class="emotion-1 emotion-2"
          color="#f3f3f3"
          id="heading5"
        >
          test
        </h5>
        <p
          class="emotion-3 emotion-4"
          color="#f3f3f3"
          id="body2"
        >
          Lorem ipsum
        </p>
      </div>
      <ul
        class="emotion-5"
      >
        <li
          class="emotion-6"
        >
          test
        </li>
        <li
          class="emotion-6"
        >
          test1
        </li>
      </ul>
    </div>
  </div>
</body>
`;
